erts/emulator/beam/utils.c

```diff
@@ -5208,11 +5208,8 @@ erts_qsort_partion_array(byte *base,
     byte* second_part_start = base + (nr_of_items * item_size);
     byte* curr = base + item_size;
     int found_bigger = 0;
-    size_t pivot_index =
-        (size_t)erts_sched_local_random_hash_64_to_32_shift((Uint64)extra_seed +
-                                                            (Uint64)((UWord)compare) +
-                                                            (Uint64)((UWord)base) -
-                                                            (Uint64)nr_of_items) % nr_of_items;
+    size_t pivot_index = nr_of_items / 2;
```

Comment:
@sverker I don't think this change is safe because the input can be designed to make the recursion depth O(N) instead of O(log(N)), if the pivot is not randomly selected.

Reply:
Ok, maybe we should find a cheaper way to do it then, because it was quite a performance gain when I removed the random pivot.

Reply:
Fixed